### DISCUSSION TITLE
GATED REVERB TYPE

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#631] [Ideas]
A 'gated' reverb, that would stop as soon as any input signal terminates. Could also be controlled with an envelop- possibly it's own envelop.
'Mama' by Genesis and the drum break in 'Jack & Dianne' By John Cougar Mellencamp made this effect a staple of 80's rock-n-roll.
Someone is already working on new reverb types, I figured this would fit into that project, assuming they were willing to take-up the task.

**Comment by ok-reza** [comment]
have you had any experience adding a large reverb to a snare, resampling it, then using that reverbed snare set to CUT mode, that way you can choose the length of the gated reverb based on when you want it to cut out manually?

experimenting with that may be useful in understanding how a gated reverb FX were to be implemented, or if the method mentioned above is sufficient in catering to the issue

**Reply by worker-gray** [reply]
I used to run my drum machine through the 'gate reverb' patch on my Yamaha SPX-50D. That's the extent of my experience.

**Reply by entzmingerc** [reply]
An envelope follower with a low threshold triggering value could achieve this. I think next year someone (hopefully me) is going to implement an envelope follower mod source, probably based on some of the compressor code. I had similar ideas to you <https://github.com/SynthstromAudible/DelugeFirmware/discussions/579> to put the code underneath shortcuts, I was thinking to use the sidechain shortcuts but underneath the envelope for different envelope shapes (hold, looping, etc) is a cool idea too.

**Reply by worker-gray** [reply]
An envelope Follower would be an excellent addition!
